src/types/Notification.ts

```diff
@@ -62,11 +56,13 @@ export interface BaseNotification extends Notification {
  * Notification triggered by a document change.
  * (create, update, delete)
  */
-export interface DocumentNotification extends BaseNotification {
+export interface DocumentNotification<
+  DocContent extends KDocumentContentGeneric = KDocumentContentGeneric
```

Comment:
We usually use a capital T to distinguish template types, can you rename it to `TDocContent` or `TContent` please?